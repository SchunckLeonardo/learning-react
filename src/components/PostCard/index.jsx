export const PostCard = (props) => {
    const { title, cover, body } = props.infos

    return (
        <div className='post'>
            <img src={cover} alt={title} />
            <div className='post-content'>
                <h1>{title}</h1>
                <p>{body}</p>
            </div>
        </div>
    )
}