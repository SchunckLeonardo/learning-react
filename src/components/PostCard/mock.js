export const postCardPropsMock = {
    title: 'post-title',
    body: 'post-body',
    cover: 'img/img.png'
}