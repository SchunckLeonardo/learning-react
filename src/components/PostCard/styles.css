.post {
    background: #eee;
    box-shadow: 0 0 10px rgba(155, 155, 155, 0.5);
    transition: transform 100ms ease-in-out;
}

.post img {
    max-width: 100%;
}

.post:hover {
    transform: scale(1.02);
}

.post-content {
    padding: 30px;
}
