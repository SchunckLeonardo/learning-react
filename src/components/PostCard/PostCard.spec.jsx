import { render, screen } from "@testing-library/react"
import { postCardPropsMock } from './mock.js'
import PostCard from '.'

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard {...postCardPropsMock} />)

        expect(screen.getByRole('img', {title: 'post-title'})).toHaveAttribute('src', postCardPropsMock.cover)
        expect(screen.getByRole('heading')).toBeInTheDocument()
        expect(screen.getByText('post-body')).toBeInTheDocument()
    })

    it('should match snapshot', () => {
        const {container} = render(<PostCard {...postCardPropsMock} />)
        expect(container).toMatchSnapshot()
    })
})