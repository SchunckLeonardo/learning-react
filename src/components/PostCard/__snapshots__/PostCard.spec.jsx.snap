// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="post-title"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        post-title
      </h2>
      <p>
        post-body
      </p>
    </div>
  </div>
</div>
`;
